docs/proposals/19-migrating-to-prometheus-operator.md

```diff
@@ -0,0 +1,529 @@
+  be missing, but others may be added.
+
+## Proposal
+
+Today, Gardener provides monitoring for shoot clusters (i.e. system components
+and the control plane) and for the seed cluster. The proposal is to migrate
+these monitoring stacks to use the [prometheus-operator]. The proposal is lined
+out below:
+
+### API
+
+Use the [API] provided by the [prometheus-operator] and create Go structs.
+
+### Prometheus Operator CRDs
+
+Deploy the [prometheus-operator] and its CRDs. These components can be deployed
+via `ManagedResources`. The operator itself and some other components outlined
+in the GEP will be deployed in a new namespace called `monitoring`. The CRDs for
+the [prometheus-operator] and the operator itself can be found
+[here][prom-crd-bundle]. This step is a prerequisite for all other steps.
+
+### Shoot Monitoring
+
+Gardener will create a monitoring stack similar to the current one with the
+[prometheus-operator] custom resources.
+
+1. Most of the shoot monitoring is deployed via this
+    [chart][shoot-monitoring]. The goal is to create a similar stack, but not
+    necessarily with feature parity, using the [prometheus-operator].
+
+    - An example Prometheus object that would be deployed in a shoot's
+      control plane.
+
+    ```yaml
+    apiVersion: monitoring.coreos.com/v1
+    kind: Prometheus
+    metadata:
+      labels:
+        app: prometheus
+      name: prometheus
+      namespace: shoot--project--name
+    spec:
+      enableAdminAPI: false
+      logFormat: logfmt
+      logLevel: info
+      image: image:tag
+      paused: false
+      portName: web
+      replicas: 1
+      retention: 30d
+      routePrefix: /
+      serviceAccountName: prometheus
+      serviceMonitorNamespaceSelector:
+        matchExpressions:
+        - key: kubernetes.io/metadata.name
+          operator: In
+          values:
+          - monitoring
+          - shoot--project--name
+      podMonitorNamespaceSelector:
+        matchExpressions:
+        - key: kubernetes.io/metadata.name
+          operator: In
+          values:
+          - monitoring
+          - shoot--project--name
+      ruleNamespaceSelector:
+        matchExpressions:
+        - key: kubernetes.io/metadata.name
+          operator: In
+          values:
+          - monitoring
+          - shoot--project--name
+      serviceMonitorSelector:
+        matchLabels:
+          monitoring.gardener.cloud/monitoring-target: seed
+      podMonitorSelector:
+        matchLabels:
+          monitoring.gardener.cloud/monitoring-target: seed
+      storage:
+        volumeClaimTemplate:
+          spec:
+            accessModes:
+            - ReadWriteOnce
+            resources:
+              requests:
+                storage: 20Gi
+      version: v2.35.0
+    ```
+
+1. Contract between the shoot `Prometheus` and its configuration.
+
+    - `Prometheus` can discover `*Monitors` in different namespaces and also
+      by using labels.
+
```

Comment:
```suggestion
```

I think this sentence could be omitted now. (Not much GEP specific added value.)